js/checkboxgroup/checkboxgroup.test.ts

```diff
@@ -0,0 +1,345 @@
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+
+		component.value = ["choice_one"];
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+	});
+});
+
+describe("Events", () => {
+	test("changing the value via the UI emits a change event", async () => {
+		const { getByLabelText, listen } = await render(CheckboxGroup, {
+			loading_status,
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+
+		const mock = listen("change");
+
+		await event.click(item_one);
+		expect(mock.callCount).toBe(1);
+
+		await event.click(item_three);
+		expect(mock.callCount).toBe(2);
+	});
+
+	test("changing the value from outside emits a change event", async () => {
+		const { getByLabelText, component, listen } = await render(CheckboxGroup, {
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+
+		const mock = listen("change");
+
+		await (component.value = [1]);
+		expect(mock.callCount).toBe(1);
+	});
+
+	test("changing the value from the UI emits an input event", async () => {
+		const { getByLabelText, listen } = await render(CheckboxGroup, {
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+
+		const mock = listen("input");
+		await event.click(item_one);
+
+		expect(mock.callCount).toBe(1);
+	});
+
+	test("changing the value from outside DOES NOT emit an input event", async () => {
+		const { getByLabelText, component, listen } = await render(CheckboxGroup, {
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+
+		const mock = listen("input");
+		await (component.value = [1]);
+
+		expect(mock.callCount).toBe(0);
+	});
+});
+
+describe("interactive vs static", () => {
```

Comment:
And then we have interactive and static. We need to test these in each component because exactly what interactive and static means is different for each component and checking the functionality is also specific.

We need more of these test, we have had quite a few issues with this behaviour just as a result of human error when refactoring.